app/src/main/java/com/example/firebasetutorial/LoginActivity.java

```diff
@@ -0,0 +1,81 @@
+package com.example.firebasetutorial;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+
+public class LoginActivity extends AppCompatActivity {
+
+    EditText etEmail, etPassword;
+    TextView goToRegistration;
+    FirebaseAuth auth;
+    Button btnLogin;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+
+        auth = FirebaseAuth.getInstance();
+
+        etEmail = findViewById(R.id.et_login_email);
+        etPassword = findViewById(R.id.et_login_password);
+        goToRegistration = findViewById(R.id.tv_go_to_register);
+        btnLogin = findViewById(R.id.login_button);
+
+        btnLogin.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                String sEmail = etEmail.getText().toString().trim();
+                String sPass = etPassword.getText().toString().trim();
+                validateFieldsAndLogin(sEmail, sPass);
+            }
+        });
+
+        goToRegistration.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                startActivity(new Intent(LoginActivity.this, RegistrationActivity.class));
+            }
+        });
+
+    }
+
+    public void validateFieldsAndLogin(String email, String pass) {
+        if (email.equals("")) {
+            etEmail.setError("This field is required");
+            return;
+        }
+        if (pass.equals("")) {
+            etPassword.setError("This field is required");
+            return;
+        }
+
+        auth.signInWithEmailAndPassword(email, pass).addOnSuccessListener(new OnSuccessListener<AuthResult>() {
+            @Override
+            public void onSuccess(AuthResult authResult) {
+                Toast.makeText(getApplicationContext(), "User logged in successfully", Toast.LENGTH_SHORT).show();
```

Comment:
Change the messgae to "Welcome" or something ?

Reply:
Okay